### DISCUSSION TITLE
How to combine two HttpService in http4s 0.15.16

### DISCUSSION THREAD

**Discussion by sujit-baniya** [http4s/http4s#6994] [Q&A] [answered]
I have a legacy product that's using http4s 0.8 version. I'm trying to do an incremental upgrade and the best version I could upgrade is 0.15.16.

Currently I could use following syntax

```scala
service1 orElse service2 orElse
```

But in 0.15.16 `orElse` is not available and I'm not sure how could I concat multiple http service.

Code sample

```scala

object Test {
	val db = MySQLAdapter(url = "jdbc:mysql:///test?useSSL=false", user = "service", password = "")
	val userDao = UserDao(db)
}

object HealthApi extends ResponseCodec with BooleanCodec {
	
	def service =
		HttpService {
			case GET -> Root / "ping" => true
		}
	
}
object Main extends ServerApp with ResponseCodec with TimestampCodec {
	def service: HttpService = HttpService {
		case GET -> Root / "hello" / name => s"Hello $name"
		case GET -> Root / "user" => Test.userDao.testUser("admin@exam.com")
	}
	def server(args: List[String]): Task[Server] =
		BlazeBuilder
			.withNio2(true)
			.bindHttp(8080)
			.mountService(service, "/api")
			.start
}
```

**Comment by armanbilge** [comment, marked as answer]
I found some old docs here: https://http4s.org/v0.15/service/index.html#running-your-service

It looks like it still works with `orElse`. Maybe you are missing `import org.http4s.server.syntax._` ?

**Reply by sujit-baniya** [reply]
@armanbilge Thanks a lot, that saved a lot.

BTW, further moving ahead if any questions arises while upgrade, should I use same discussion on github or there's any chat server available?

**Reply by armanbilge** [reply]
Here is fine, there is also a Discord chat :)
https://discord.gg/beMhYbC5tM
